tests/test_requests_batch.py

```diff
@@ -375,3 +532,134 @@ def test_code():
             assert execution_results[0].extensions["key1"] == "val1"
 
     await run_sync_test(event_loop, server, test_code)
+
+
+@pytest.mark.online
+@pytest.mark.requests
+def test_requests_sync_batch_auto():
+
+    from threading import Thread
+    from gql.transport.requests import RequestsHTTPTransport
+
+    client = Client(
+        transport=RequestsHTTPTransport(url="https://countries.trevorblades.com/"),
+        batch_interval=0.01,
+        batch_max=3,
+    )
+
+    query = gql(
+        """
+        query getContinentName($continent_code: ID!) {
+          continent(code: $continent_code) {
+            name
+          }
+        }
+        """
+    )
+
+    def get_continent_name(session, continent_code):
+        variables = {
+            "continent_code": continent_code,
+        }
+
+        result = session.execute(query, variable_values=variables)
+
+        name = result["continent"]["name"]
+        print(f"The continent with the code {continent_code} has the name: '{name}'")
+
+    continent_codes = ["EU", "AF", "NA", "OC", "SA", "AS", "AN"]
+
+    with client as session:
+
+        for continent_code in continent_codes:
+
+            thread = Thread(
+                target=get_continent_name,
+                args=(
+                    session,
+                    continent_code,
+                ),
+            )
+            thread.start()
+
+    # Doing it twice to check that everything is closing and reconnecting correctly
+    with client as session:
+
+        for continent_code in continent_codes:
+
+            thread = Thread(
+                target=get_continent_name,
+                args=(
+                    session,
+                    continent_code,
+                ),
+            )
+            thread.start()
```

Comment:
don't we should join the thread after it's started?

Reply:
I don't think it's really necessary but it does not hurt, so I changed it.